### ISSUE TITLE
Moving location of "Snippets" plug-in's input dialog box

### ISSUE THREAD

**Issue by paul1149** [hluk/CopyQ#1275]
Hi folks,
Still using and enjoying CopyQ!

I used the Snippets plug in a lot, and because of a bug in the XFCE desktop's hover-focus, as soon as I dismiss its input dialog the window beneath it gets immediate focus. So the snippet is posted to that window rather than the last window I was actually using.

Therefore, I'd like to move the Snippet's dialog a bit to the right, so it is over the LibreOffice password dialog that I am trying to populate. Currently Snippets is opening about 50px off the left edge of the desktop.

I see in the documentation that dialog has these parameters:

`‘.height’, ‘.width’, ‘.x’, ‘.y’ - dialog geometry`

but I don't know how to implement them. Any help would be much appreciated.

Be well!

**Comment by hluk** [comment]
For the [snippets command](https://github.com/hluk/copyq-commands/blob/master/Global/snippets.ini) you can set those arguments for `dialog()` calls. Specifically, these parts:

```js
      var snippet = dialog(
        '.title', 'Select Snippet',
        list + 'Snippet', snippetNames
      ) || abort()
...
      var dialogVars = [
        '.title', 'Set Snippet Values',
        '.label', label
      ]
```

So under the `.title` lines you could add something like:

```
    '.x', 1850,
    '.y', 1000,
```

**Comment by paul1149** [comment]
Thanks, hluk. I'm having problems knowing where to place that. I've tried:

`function askForSnippet(snippetNames, snippets) {
  var list = listSnippets ? '.list:' : ''
  var snippet = dialog(
    '.title', 'Select Snippet',
    list + 'Snippet', snippetNames,
    '.x.', 1300
  ) || abort()
`
I've tried:

`function askForSnippet(snippetNames, snippets) {
  var list = listSnippets ? '.list:' : ''
  var snippet = dialog(
    '.title', 'Select Snippet',
    list + 'Snippet', snippetNames
  ) || abort()
  var dialogVars = [
      '.x.' 300
  ]
`

All to no avail.

**Comment by hluk** [comment]
It could be something like this: https://gist.github.com/hluk/3041da1814b91f6daa9a951b4e1d49c6

**Comment by paul1149** [comment]
Thank you, Lukas. Sorry, but I'm having no success. Did you intend for the script to grow from 130 to 313 lines?

**Comment by hluk** [comment]
@paul1149 Oh, sorry, not sure what happened, looks like the command was incorrectly pasted or saved in the gist. Fixed: https://gist.github.com/hluk/3041da1814b91f6daa9a951b4e1d49c6

**Comment by paul1149** [comment]
Thanks again, Lukas. Unfortunately nothing has been able to move that combo box yet. I'm on MX Linux 19 x64, which is based on Debian 'buster'.

**Comment by paul1149** [comment]
I brought the hover-focus bug up to xfce, btw, quite a while ago, but there doesn't seem to be interest there in dealing with it. But I suppose the ability to move this box is a good thing anyway.

**Comment by paul1149** [comment]
Hello,
I wanted to add that I have solved this problem, and in a far more organic way.  I simply added a sleep function before pasting. Apparently the speed of the script was giving the computer problems.  Now it doesn't matter where the dialog is placed, the system's window control has enough time to refocus on the target application.
Thanks again.
Paul

```
function pasteSnippet(mime, content) {
  copy(mime, content)
  copySelection(mime, content)
  sleep(500)
  paste()
}
```
